qual/src/main/java/com/virtuslab/qual/guieffect/UIThreadUnsafe.java

```diff
@@ -6,6 +6,8 @@
 import java.lang.annotation.Target;
 
 // TODO (typetools/checker-framework#3252): replace with proper @Heavyweight
-@Retention(RetentionPolicy.CLASS)
-@Target({ElementType.METHOD})
-public @interface NotUIThreadSafe {}
+
+// Must have runtime retention to be visible to ArchUnit tests.
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.METHOD, ElementType.CONSTRUCTOR})
+public @interface UIThreadUnsafe {}
```

Comment:
I don't like this name TBH, mostly due to the negation in _unsafe_ which makes things confusing... any ideas how to rename? 🤔